lib/inspec/input_registry.rb

```diff
@@ -166,15 +166,44 @@ def bind_inputs_from_cli_args(profile_name, input_list)
           end
         end
         input_name, input_value = pair.split("=")
+        input_value = clean_up_cli_input_value(input_name, input_value)
         evt = Inspec::Input::Event.new(
-          value: input_value.chomp(","), # Trim trailing comma if any
+          value: input_value,
           provider: :cli,
           priority: 50
         )
         find_or_register_input(input_name, profile_name, event: evt)
       end
     end
 
+    # Remove trailing commas, resolve type.
+    def clean_up_cli_input_value(input_name, given_value)
```

Comment:
Suggest `value_from_cli_input` or `parse_cli_input_value` or some other name that expresses that this method evaluates a value - the current `clean_up_` name suggests it just trims or does simple casting.